src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -313,7 +313,6 @@ private Connection newConnection() throws Exception
 
          connection = (username == null) ? dataSource.getConnection() : dataSource.getConnection(username, password);
          setupConnection(connection);
-         lastConnectionFailure.set(null);
```

Comment:
Reporting an old exception, after the database has already recovered, can be very confusing for the user.

If the getConnection() call times out because all connections are in use, but reports a communications failure, we've mislead the user -- especially because we know that the database has recovered.

I am prepared to merge this request, but not with this change.